master/jobmanager.go

```diff
@@ -15,109 +15,100 @@ import (
 	"go.uber.org/zap"
 )
 
-// JobManager manages all the job masters, and notify the offline executor to them.
+// JobMasterID is special and has no use.
+const JobMasterID model.ID = -1
+
+// JobManager is a special job master that manages all the job masters, and notify the offline executor to them.
 type JobManager struct {
-	mu         sync.Mutex
-	jobMasters map[model.JobID]system.JobMaster
+	*system.Master
 
+	mu          sync.Mutex
+	jobMasters  map[model.ID]*model.Task
 	idAllocater *autoid.IDAllocator
-
-	offExecutors chan model.ExecutorID
-
+	clients     *client.Manager
 	masterAddrs []string
 }
 
-// Start the deamon goroutine.
-func (j *JobManager) Start(ctx context.Context) {
-	go j.startImpl(ctx)
-}
-
-func (j *JobManager) startImpl(ctx context.Context) {
-	for {
-		select {
-		case execID := <-j.offExecutors:
-			log.L().Logger.Info("notify to offline exec")
-			j.mu.Lock()
-			for _, jobMaster := range j.jobMasters {
-				jobMaster.OfflineExecutor(execID)
-			}
-			j.mu.Unlock()
```

Comment:
Will the offline logic be added in another PR?

Reply:
I think so, I should be added after we make sure the logic of heartbeats